File: android/app/src/main/java/com/cuhk/travelligent/HomeActivity.java
package com.cuhk.travelligent;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v4.app.FragmentManager;
import android.view.View;
import android.support.design.widget.NavigationView;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.TextView;

import com.cuhk.travelligent.sight.SightDetailsFragment;
import com.cuhk.travelligent.sight.SightFragment;

import io.swagger.client.models.SightListDto;

import static com.cuhk.travelligent.sight.SightDetailsFragment.SightDetails_SightId;

public class HomeActivity extends AppCompatActivity
        implements NavigationView.OnNavigationItemSelectedListener, SightFragment.OnListFragmentInteractionListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });

        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        drawer.addDrawerListener(toggle);
        toggle.syncState();

        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
        View headerView = navigationView.getHeaderView(0);
        navigationView.setNavigationItemSelectedListener(this);

        SharedPreferences prefs = getSharedPreferences(Configs.PREFS, MODE_PRIVATE);

        TextView nameView = headerView.findViewById(R.id.name_view);
        nameView.setText(prefs.getString(Configs.PREFS_FIRST_NAME, "FIRST") + " " + prefs.getString(Configs.PREFS_LAST_NAME, "FIRST"));

        TextView emailAddressView = headerView.findViewById(R.id.email_address_view);
        emailAddressView.setText(prefs.getString(Configs.PREFS_EMAIL_ADDRESS, "EMAIL_ADDRESS"));
    }

    @Override
    public void onBackPressed() {
        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START);
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.home, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @SuppressWarnings("StatementWithEmptyBody")
    @Override
    public boolean onNavigationItemSelected(MenuItem item) {
        // Handle navigation view item clicks here.
        int id = item.getItemId();

        if (id == R.id.nav_sights) {
            setTitle("Sights");
            SightFragment sightFragment = new SightFragment();
            FragmentManager fragmentManager = getSupportFragmentManager();
            fragmentManager.popBackStackImmediate(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);
            fragmentManager.beginTransaction().replace(R.id.homeFragment, sightFragment).commit();
        }

        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        drawer.closeDrawer(GravityCompat.START);
        return true;
    }

    @Override
    public void onListFragmentInteraction(SightListDto item) {
        System.out.println(item);

        setTitle(item.getTitle());

        SightDetailsFragment sightDetailsFragment = new SightDetailsFragment();
        Bundle args = new Bundle();
        args.putInt(SightDetails_SightId, item.getId());
        sightDetailsFragment.setArguments(args);

        FragmentManager fragmentManager = getSupportFragmentManager();
        fragmentManager.popBackStackImmediate(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);
        fragmentManager.beginTransaction().replace(R.id.homeFragment, sightDetailsFragment).commit();
    }
}

File: android/app/src/main/java/com/cuhk/travelligent/HomeActivity.kt
package com.cuhk.travelligent

import android.content.Context
import android.os.Bundle
import androidx.core.view.GravityCompat
import androidx.appcompat.app.ActionBarDrawerToggle
import android.view.MenuItem
import androidx.drawerlayout.widget.DrawerLayout
import com.google.android.material.navigation.NavigationView
import androidx.fragment.app.FragmentManager
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import android.view.Menu
import androidx.appcompat.widget.SearchView
import com.cuhk.travelligent.page.sights.SightsFragment
import kotlinx.android.synthetic.main.nav_header_home.view.*
import androidx.core.view.MenuItemCompat
import com.cuhk.travelligent.page.hotels.HotelsFragment


class HomeActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        val toolbar: Toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)

        val drawerLayout: DrawerLayout = findViewById(R.id.drawer_layout)
        val navView: NavigationView = findViewById(R.id.nav_view)
        val toggle = ActionBarDrawerToggle(
            this, drawerLayout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close
        )
        drawerLayout.addDrawerListener(toggle)
        toggle.syncState()

        // Nav
        navView.setNavigationItemSelectedListener(this)

        val headerView = navView.getHeaderView(0)
        val nameView = headerView.name
        val emailAddressView = headerView.email_address

        val prefs = getSharedPreferences(Configs.PREFS, MODE_PRIVATE)
        nameView.text = prefs.getString(
            Configs.PREFS_FIRST_NAME,
            "FIRST"
        ) + " " + prefs.getString(Configs.PREFS_LAST_NAME, "FIRST")
        emailAddressView.text = prefs.getString(Configs.PREFS_EMAIL_ADDRESS, "EMAIL_ADDRESS")

        // Default
        supportFragmentManager
            .popBackStackImmediate(null, FragmentManager.POP_BACK_STACK_INCLUSIVE)

        val sightsFragment = SightsFragment()
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.home_fragment, sightsFragment)
            .commit()

    }

    override fun onBackPressed() {
        val drawerLayout: DrawerLayout = findViewById(R.id.drawer_layout)
        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_search -> true
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        when (item.itemId) {
            R.id.nav_home -> {
                supportFragmentManager
                    .popBackStackImmediate(null, FragmentManager.POP_BACK_STACK_INCLUSIVE)

                val sightsFragment = SightsFragment()
                supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.home_fragment, sightsFragment)
                    .commit()
            }
            R.id.nav_hotel -> {
                supportFragmentManager
                    .popBackStackImmediate(null, FragmentManager.POP_BACK_STACK_INCLUSIVE)

                val sightsFragment = HotelsFragment()
                supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.home_fragment, sightsFragment)
                    .commit()
            }
            R.id.nav_log_out -> {
                val editor = getSharedPreferences(Configs.PREFS, Context.MODE_PRIVATE).edit()
                editor.clear()
                editor.apply()

                finish()
            }
        }

        val drawerLayout: DrawerLayout = findViewById(R.id.drawer_layout)
        drawerLayout.closeDrawer(GravityCompat.START)

        return true
    }
}
